### ISSUE TITLE
Lombok installer cannot find eclipse MARS on Mac OS X 10.10

### ISSUE THREAD

**Issue by lombokissues** [rzwitserloot/lombok#864]
_Migrated from Google Code (issue 829)_

**Comment by jbcpollak** [comment]
Agreed, I have this problem as well. When I clicked "Specify location', the file browser would not let me accept the 'Eclipse.app' file, instead it opened as if it was a folder. I had to browse to the eclipse.ini file. I have Eclipse installed in a non-standard location, so I didn't expect lombok to find it, but I did expect it to recognize the package.

**Comment by marcelstoer** [comment]
+1

**Comment by kftweddle** [comment]
I got it to install in eclipse.ini, but <b>eclipse -clean</b> reports:
<code>
Error opening zip file or JAR manifest missing : lombok.jar
<p/>
Error occurred during initialization of VM
</code>
...after which it fails to launch.

**Comment by vijaydev** [comment]
I was using lombok 1.12 with Eclipse Mars and had the same problem. I copied the jar manually into /Applications/Eclipse.app/Contents/MacOS as lombok.jar and then Eclipse started fine. 

But then, I saw http://stackoverflow.com/a/22332248/27474 and decided to try with the latest lombok (1.16.6) and it worked perfectly with Mars.

**Comment by cristcost** [comment]
Launching the installer on my mac only finds installation of Eclipse Luna and Kepler:

<img width="760" alt="schermata 2016-01-21 alle 17 37 52" src="https://cloud.githubusercontent.com/assets/2135798/12487101/04955238-c066-11e5-95bb-135dfc410fa8.png">

Googling I've found that in order to have the installer see the installation of Eclipse Mars, you need to navigate to the "eclipse.ini" file in the Eclipse.app/Contents/Eclipse folder.
<img width="713" alt="schermata 2016-01-21 alle 17 38 26" src="https://cloud.githubusercontent.com/assets/2135798/12487188/65f7e806-c066-11e5-9c77-b8f5688132fa.png">

Then the installer would work:
<img width="765" alt="schermata 2016-01-21 alle 17 38 41" src="https://cloud.githubusercontent.com/assets/2135798/12487201/6f95494e-c066-11e5-8d05-ad68ed55eca8.png">

Maybe something has changed on the layout of the mac application of eclipse.

It would be nice if in the next version of the installer it is able to recognize also Eclipse Mars installations.

**Comment by jingxuan1990** [comment]
it doesn't work. i can't install it for my eclipse neno.

**Comment by gigaSproule** [comment]
Finding the .ini file works for me with neon

**Comment by kickkick** [comment]
Hi.
I succeeded adding lombok after selecting eclipse.ini in the lombok installer.
Eclipse, however, cannot start up , saying like below.

$ ./Contents/MacOS/eclipse -clean
Error occurred during initialization of VM
Could not find agent library instrument on the library path, with error: dlopen(libinstrument.dylib, 1): image not found
$ 

Here's My environment.
OS : MacOS EI Caption
Eclipse : Neon.1 Release(4.6.1)

If you could help me, it would be great.

**Comment by bittracer** [comment]
@gigaSproule Can you elaborate as I am on Mac and using neon, it does not work.

**Comment by rspilker** [comment]
Can you please try the [instructions on stackoverflow](http://stackoverflow.com/questions/42380528/lombok-not-working-with-eclipse-neon-2-on-macos-sierra/42401588#42401588) and report if they work for you?

**Comment by gigaSproule** [comment]
@bittracer If you run the jar, then click `Specify location...`. In the file dialogue, find your Eclipse installation (`Eclipse.app`), go into it, then go into `Contents` > `Eclipse` and open `eclipse.ini`. That should be it.

**Comment by rspilker** [comment]
Neon on Mac should be found automatically from the next release: https://github.com/rzwitserloot/lombok/commit/ad4bd36274d2bfe8dcf74977d921a114739386b1

**Comment by bittracer** [comment]
It worked in eclipse Luna on Mac OS Sierra.@gigaSproule  I although tried the same for Eclipse Neon.2 & it seems no success. I have also given absolute path for lombok.jar yet, it did not worked on Mac. @rspilker Will wait until next release & till then I am good to go with Luna.   

Thanks for Quick Response !

**Comment by rspilker** [comment]
I see that you're using the oomph installer. We did encounter some problems with it. So from the next release on, we'll always specify an absolute path.
